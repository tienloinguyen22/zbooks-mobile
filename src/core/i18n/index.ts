import i18next, { Module, Services, InitOptions } from 'i18next';
import { initReactI18next } from 'react-i18next';
import en from './locales/en.json';
import vi from './locales/vi.json';

const LANGUAGE_VI = 'vi';
const LANGUAGE_EN = 'en';
export type LanguageType = typeof LANGUAGE_EN | typeof LANGUAGE_VI;
export interface Language {
  id: string;
  name: string;
}

interface LanguageDetectorAsyncModule extends Module {
  type: 'languageDetector';
  /** Set to true to enable async detection */
  async: true;
  init(services: Services, detectorOptions: object, i18nextOptions: InitOptions): void;
  /** Must call callback passing detected language */
  detect(callback: (lng: string) => void): void;
  cacheUserLanguage(lng: string): void;
}

export const LANGUAGES = [
  {
    id: LANGUAGE_EN,
    name: 'English',
  },
  {
    id: LANGUAGE_VI,
    name: 'Tiếng Việt',
  },
];

const getLanguageName = (id: LanguageType): string | undefined => {
  const language = LANGUAGES.find((lang) => lang.id === id);
  return language ? language.name : undefined;
};

const getLanguageByName = (name: string): Language | undefined => LANGUAGES.find((lang) => lang.name === name);

const LANGUAGE_TEXTS = LANGUAGES.map((lang) => lang.name);

const initialize = async (): Promise<void> => {
  const languageDetector: LanguageDetectorAsyncModule = {
    type: 'languageDetector',
    async: true,
    detect: (cb: (language: string) => void) => cb(LANGUAGE_EN),
    init: () => {
      // blank
    },
    cacheUserLanguage: () => {
      // blank
    },
  };

  await i18next
    .use(languageDetector)
    .use(initReactI18next)
    .init({
      fallbackLng: LANGUAGE_EN,
      debug: __DEV__,
      resources: {
        en: {
          translation: en,
        },
        vi: {
          translation: vi,
        },
      },
    });
};

export const i18n = {
  LANGUAGE_VI,
  LANGUAGE_EN,
  LANGUAGES,
  LANGUAGE_TEXTS,
  getLanguageName,
  getLanguageByName,
  initialize,
};
