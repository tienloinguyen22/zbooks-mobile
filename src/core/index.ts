export * from './i18n';
export * from './themes/colors';
export * from './interfaces/ScreenProps';
export * from './interfaces/LoginType';
export * from './interfaces/User';
export * from './interfaces/Resource';
export * from './interfaces/Theme';
export * from './helpers/show_notification';
export * from './helpers/sleep';
export * from './helpers/record_error';
export * from './helpers/configure_google_sign_in';
export * from './helpers/combine_styles';
export * from './helpers/configure_google_sign_in';
export * from './helpers/handle_error';
export * from './helpers/get_primary_color';
export * from './helpers/format_date';
export * from './screen_names';
