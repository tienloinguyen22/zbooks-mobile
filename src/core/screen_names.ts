export const screenNames = {
  LoginScreen: 'LoginScreen',
  EmailRegisterScreen: 'EmailRegisterScreen',
  EmailLoginScreen: 'EmailLoginScreen',
  PhoneNoLoginScreen: 'PhoneNoLoginScreen',
  EmailVerificationScreen: 'EmailVerificationScreen',
  ChangePasswordScreen: 'ChangePasswordScreen',
  ForgotPasswordScreen: 'ForgotPasswordScreen',
  AppLoaderScreen: 'AppLoaderScreen',
  SettingsScreen: 'SettingsScreen',
  HomeScreen: 'HomeScreen',
  NewScreen: 'NewScreen',
  FinishRegisterScreen: 'FinishRegisterScreen',
  FavoriteBooksScreen: 'FavoriteBooksScreen',
  UserProfileScreen: 'UserProfileScreen',
  EditEmailScreen: 'EditEmailScreen',
  EditFullnameScreen: 'EditFullnameScreen',
  EditPhoneNoScreen: 'EditPhoneNoScreen',
  EditGenderScreen: 'EditGenderScreen',
};
