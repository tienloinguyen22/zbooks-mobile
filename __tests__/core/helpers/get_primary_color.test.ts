import { getPrimaryColor } from '@app/core';

describe('core/helpers/get_primary_color', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });
  it('returns default color if not found', async () => {
    const color = getPrimaryColor('notFoundColor', 'light');
    expect(color).toMatchSnapshot();
  });

  it('returns default light color if the current theme is light', async () => {
    const color = getPrimaryColor('red', 'light');
    expect(color).toMatchSnapshot();
  });

  it('returns default light color if the current theme is dark', async () => {
    const color = getPrimaryColor('red', 'dark');
    expect(color).toMatchSnapshot();
  });
});
