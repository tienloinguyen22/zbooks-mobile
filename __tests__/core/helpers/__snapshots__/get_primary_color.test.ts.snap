// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core/helpers/get_primary_color returns default color if not found 1`] = `"#fff"`;

exports[`core/helpers/get_primary_color returns default light color if the current theme is dark 1`] = `"#e57373"`;

exports[`core/helpers/get_primary_color returns default light color if the current theme is light 1`] = `"#d32f2f"`;
